python/lsst/pipe/base/tests/util.py

```diff
@@ -111,3 +118,27 @@ def get_output_refs(graph: QuantumGraph) -> list[DatasetRef]:
             result += [ref for ref in init_refs if ref in output_refs]
 
     return result
+
+
+@contextmanager
+def patch_deterministic_uuid4(seed: int) -> Iterator[None]:
```

Comment:
Would these functions be better off in lsst.utils?  For example, in case ctrl_bps provenance is asked to use UUID.